### ISSUE TITLE
Label becomes blank If a large text is posted in it. 

### ISSUE THREAD

**Issue by varshneyshikhar** [cbpowell/MarqueeLabel#96]
if I paste a text as as large as this:-
" Raiders, Michael Crabtree agree to terms on reported 1-year, $3-million deal  Ex-NFL RB Lawrence Phillips suspected of killing prison cellmate   Bears' Matt Forte not present for start of voluntary workouts   Adrian Peterson unsure about participating in Vikings' offseason workouts if reinstated   Dolphins reportedly remain focused on extension for Ryan Tannehill   William & Mary takes Darren Sharper out of Hall of Fame   Eagles sign Cedric Thornton to 1-year contract   Report: Hakeem Nicks visiting Titans   Broncos' Brandon Marshall signs exclusive rights tender   Broncos' Emmanuel Sanders: 'If we don't bring a Super Bowl ... it's a bust season'  VIDEO: Tom Brady throws out 1st pitch at Red Sox home opener   Report: Lions dispute Buccaneers' offer sheet to George Johnson Bills OC dfdfhfdygfhfh ffer = 30.0f Raiders, Michael Crabtree agree to terms on reported 1-year, $3-million deal  Ex-NFL RB Lawrence "

Your every label becomes blank and stops working

**Comment by cbpowell** [comment]
I think this is similar to issue #95, and unfortunately it's a length limitation of the standard UILabel (which MarqueeLabel uses internally). Thanks for giving a sample text string though, I'll check it out later today. Feel free to check it out yourself too, by setting up a standard UILabel and passing your sample string to it.

**Comment by cbpowell** [comment]
Well, I don't have any issue using your sample string in the MarqueeLabel demo project, or in a simple project I set up to test if a regular UILabel could handle it.

Could you post how you're setting up your MarqueeLabel instance?

**Comment by shikharvarshney** [comment]
Yeah sure!!
I had put this string in your demo project also and then also it was creating problems which I have mentioned above.

The code which I ma using BTW is this
-(void)setUpNewsFeedLabel:(MarqueeLabel *)newsFeedLabel{
    
    // Continuous, with tap to pause
    newsFeedLabel.marqueeType = MLContinuous;
    //self.lbl_newsFeed.scrollDuration = 10.0f;
    newsFeedLabel.rate = 50.0f;
    newsFeedLabel.fadeLength = 10.0f;
    newsFeedLabel.trailingBuffer = 100.0f;
    
    
    newsFeedLabel.userInteractionEnabled = YES; // Don't forget this, otherwise the gesture recognizer will fail (UILabel has this as NO by default)
    UITapGestureRecognizer *tapRecognizer = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(pauseTap:)];
    tapRecognizer.numberOfTapsRequired = 1;
    tapRecognizer.numberOfTouchesRequired = 1;
    [newsFeedLabel addGestureRecognizer:tapRecognizer];
    
}

and then in a separate method:-  

for (int i = 0; i<[feeds count]; i++) {
    NSString* result = [[[feeds objectAtIndex:i] valueForKey:@"title"] stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
    [arr_feedTitles addObject:result] ;
    }
    NSLog(@"the feeds array is %@", arr_feedTitles);
    NSString * str_feedSequence = [arr_feedTitles componentsJoinedByString:@" || "];
    
    NSLog(@"the string for feed sequencs is %@", str_feedSequence);
    if (!isCBSFeedLoaded) {
        self.lbl_cbsSportsnewsFeed.text = str_feedSequence;
        isCBSFeedLoaded = TRUE;
    }
    else{
        self.lbl_newsFeed.text = str_feedSequence;
 
    }

The str_feedSequence is getting logged on the console but somehow not appearing on the Marquee label.

**Comment by cbpowell** [comment]
Hmm, have you verified the label frame? I can't tell if you're using AutoLayout, but you might want to set a breakpoint on your `self.lbl_newsFeed.text = ...` line (or the CBS one, whichever) and make sure the frame doesn't have a zero height or width. 

If that doesn't reveal anything, have you tried it with test strings? I'd suggest trying at least a short string and a long string example - one that won't make MarqueeLabel scroll and one that will:
```objective-c
NSLog(@"the string for feed sequencs is %@", str_feedSequence);
// Change the str_feedSequence manually
str_feedSequence = @"This is a short string"; // something short enough NOT to cause an auto-scroll
// or...
str_feedSequence = @"An example longer string, with more content and making it wider"; // or whatever to make it long enough to auto-scroll
NSLog(@"Manual str_feedSequence: %@", str_feedSequence);
if (!isCBSFeedLoaded) {
    self.lbl_cbsSportsnewsFeed.text = str_feedSequence;
    isCBSFeedLoaded = TRUE;
}
else{
    self.lbl_newsFeed.text = str_feedSequence;
}
```
If you still have blank labels after trying manual strings, make sure your text color isn't the same as your background!

**Comment by iamabhiee** [comment]
problem with me too with string "Retired Chiefs tight end Sean McGrath reinstated from reserve list, waived || Jaelen Strong suffers fractured bone in wrist || Bryce Petty was invited to NFL draft, but will not attend || Stadium proposal for Chargers, Raiders qualifies for ballot || Dolphins guard Daryn Colledge retiring after 9 seasons || Redskins re-sign kicker Kai Forbath to 1-year, $1.54M contract || Von Miller embracing new diet, believes Broncos will be top defense || Tom Brady, Rex Ryan make light of poor 1st pitch at Red Sox game || Chargers sign Chris Hairston || Aaron Hernandez found guilty of first-degree murder in death of Odin Lloyd || Fred Jackson on LeSean McCoy: 'I'm not going to make it easy for him' || Sammy Watkins vouches for EJ Manuel as Bills' starting QB || Saints agree to terms with Mike Golic || Saints hosting Corey Wootton || Brian Westbrook not convinced LeSean McCoy didn't fit Chip Kelly's system || NFL players react to guilty verdict in Aaron Hernandez murder trial || Browns' Barkevious Mingo 'pretty close to all-go' following shoulder surgery || Ex-NFL linebacker Dhani Jones makes case for ditching hard-shell helmets || Adrian Peterson eligible for reinstatement beginning Wednesday || Tom Brady says he jumped off cliff to impress his wife ||"

**Comment by shikharvarshney** [comment]
Yes I am using auto layout and the Label is visible on the screen at runtime. For checking this effect only I have kept two Marquee labels one above the other. The one which is having short enough string is able to create the Marquee effect but the long one is not able to do this. 
See the problem here lies in the"length of the string" only. Also I have checked the text color and all that.
Also try to put the String which I have posted to you in your demo project just statically, That will show you the Error.
If you can provide me some way, then I would like to show you the video demo of the implemantation.

I am using .h and .m files from cocoacontrols and iOS 8.3
Please do look into it as other users are also facing the same problems.

Thanks

**Comment by cbpowell** [comment]
I was able to replicate the problem with the string @iamabhiee provided…unfortunately it is the same problem as issue #95, although I now know the root cause.

Your strings are so long, they're causing the UILabels used by MarqueeLabel to grow to wider than 8192 points (1024 x 8) at which point they fail to display anything. You can test this by using @iamabhiee's string in a normal UILabel, and setting the width to 8192 - the string will display. Set the width to 8193 and try again, and note that the label simply doesn't draw on the screen. MarqueeLabel currently set the width of the internal UILabels to the size required to display the entire string, so if that is larger than 8192 points you run into this issue.

I'm going to revise MarqueeLabel to simply limit the frame width of the sub-UILabels to 8192 points, and any text beyond that will be truncated. So unfortunately you'll need to find a different approach!

I will say that your use-case isn't what MarqueeLabel is designed for, although it will work (at least up to a certain length). You might want to look into something like a paging, horizontally-scrolling scroll view, where you can page in and out UILabel views for each of your feed headlines - take a look at [SwipeView](https://github.com/nicklockwood/SwipeView).

**Comment by shikharvarshney** [comment]
Thanks mate for the explanation. I would love to find a solution for this limitation as your Library is perfect tom use for my use case apart from this case. As these are multiple strings combined into one for marquee purposes, they ought to be long without any fail.
Please do update if you find any workaround fro this case.

Thanks

**Comment by cbpowell** [comment]
Implemented the width limit in [release v2.2.2](https://github.com/cbpowell/MarqueeLabel/releases/tag/2.2.2), which at least prevents blank UILabels. Unless Apple revises UILabel to draw when wider than 8,192 points, I don't think there's any other reasonable fix I can make - sorry!

**Comment by shikharvarshney** [comment]
Thanks
